pkg/cli/util.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+)
+
+func ReadContent(path string) (string, error) {
+	_, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", fmt.Errorf("file [%s] does not exist", path)
+		}
+		return "", err
+	}
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+func parseVCHost(host string) string {
+	ip := net.ParseIP(host)
+	if ip != nil {
+		return "https://username:password@" + ip.String() + "/sdk"
+	}
+	section := strings.Split(host, ":")
+	for _, v := range section {
+		if v == "https" || v == "http" {
+			continue
+		}
+		ip = net.ParseIP(v)
+		if ip != nil {
+			return "https://username:password@" + ip.String() + "/sdk"
+		}
+	}
+	return "https://username:password@host/sdk"
```

Comment:
Well, the basic idea is to initialize `UserInfo` with `username` and `password`, since the it is always required for calling rest api. In the following `processCredential()` call  the actual user and password will be loaded and overwrite. The `processCredential` call will handle the credentials from secret in the future.

Reply:
The basic idea is to initialize `UserInfo` with `username` and `password`. The following 
 `processCredential()` call will overwrite the actual credentials based on the `Credential` strategy. 
```golang
type Credential struct {
	username string
	password string
	cert     string
	Secrete  VCCMSecret
}
```
1) declarative username and password
2) Secrete in k8s

Reply:
`soap.ParseURL` can be used here instead?  ok to initialize `UserInfo`, but it should be with empty `""` username and password.